File: 2022/aoc_2022/test/day_1_test.exs
defmodule Day1Test do
  use ExUnit.Case
  doctest Aoc2022.Day1

  test "part 1-example" do
    assert Aoc2022.Day1.solve("assets/day1_example.txt") == 24000
  end

  test "part 1" do
    assert Aoc2022.Day1.solve("assets/day1.txt") == 68775
  end

  test "part 2-example" do
    assert Aoc2022.Day1.solve2("assets/day1_example.txt") == 45000
  end

  test "part 2" do
    assert Aoc2022.Day1.solve2("assets/day1.txt") == 202585
  end
end

File: 2024/aoc2024.Tests/Day1Test.cs
using System;
using System.IO;
using day_1;
using JetBrains.Annotations;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace aoc2024.Tests;

[TestClass]
[TestSubject(typeof(Day1))]
public class Day1Test
{

    [TestMethod]
    public void testSolve1Example()
    {
        var projectDirectory = Directory.GetParent(AppContext.BaseDirectory).FullName;
        var filePath = Path.Combine(projectDirectory, "inputs", "day1-example.txt");
        Assert.AreEqual(11, Day1.solve1(filePath));
    }
    
    [TestMethod]
    public void testSolve1()
    {
        var projectDirectory = Directory.GetParent(AppContext.BaseDirectory).FullName;
        var filePath = Path.Combine(projectDirectory, "inputs", "day1-input.txt");
        Assert.AreEqual(2196996, Day1.solve1(filePath));
    }
    
    [TestMethod]
    public void testSolve2Example()
    {
        var projectDirectory = Directory.GetParent(AppContext.BaseDirectory).FullName;
        var filePath = Path.Combine(projectDirectory, "inputs", "day1-example.txt");
        Assert.AreEqual(31, Day1.solve2(filePath));
    }
    [TestMethod]
    public void testSolve2()
    {
        var projectDirectory = Directory.GetParent(AppContext.BaseDirectory).FullName;
        var filePath = Path.Combine(projectDirectory, "inputs", "day1-input.txt");
        Assert.AreEqual(23655822, Day1.solve2(filePath));
    }

}